src/snakemake/api.py

```
@@ -589,6 +588,9 @@ def execute_workflow(
             if execution_settings.debug:
                 raise ApiError("debug mode cannot be used with non-local execution")
 
+        assert self.workflow_api.resource_settings.default_resources is not None
```

Ruff (0.12.2)

Use of `assert` detected

(S101)